docs/run_test/azure_auth.rst

```diff
@@ -0,0 +1,164 @@
+
+Available Authentication Methods
+-------------------------------
+1. `Default Credentials <#default-credentials>`
+2. `Certificate Authentication <#certificate-authentication>`
+3. `Assertion Authentication <#assertion-authentication>`
+5. `Workload Identity Authentication <#workload-identity-authentication>`
+6. `Token Authentication <#token-authentication>`
+7. `Client Secret Authentication <#client-secret-authentication>`
+
+Default Credentials
+-------------------
+Default authentication uses credentials from environment variables, Azure CLI, or managed identities. This method is useful when your environment is already authenticated through Azure CLI, Azure PowerShell or managed identities.
+
+Example:
+.. code:: yaml
+
+   platform:
+     - type: azure
+       azure:
+         credential:
+           type: default
+           client_id: <client id>  # Optional
+           tenant_id: <tenant id>  # Optional
+           allow_all_tenants: false | true  # Optional. Default is `false`.
+
+* **type**: `default` indicates default credential authentication.
+* **client_id**: (Optional) Needed when there are multiple managed identities associated with the running machine.
+* **tenant_id**: (Optional) Needed when you have multiple tenants.
+* **allow_all_tenants**: (Optional) Specifies whether to allow cross-tenant authorization. Default is `false`.
+
+Certificate Authentication
+---------------------------
+Authenticates as a service principal using a certificate. You will need to provide the certificate file path and optionally specify whether to send the certificate chain.
+
+Example:
+.. code:: yaml
+
+   platform:
+     - type: azure
+       azure:
+         credential:
+           type: certificate
+           tenant_id: <tenant id> # Required
+           client_id: <client id> # Required
+           cert_path: <path to certificate file> # Required
+           client_send_cert_chain: false | true # Optional. Default is `false`.
+
+* **type**: `certificate` indicates certificate-based authentication.
+* **tenant_id**: ID of the principal's tenant. Also called its "directory" ID.
+* **client_id**: The principal's client ID.
+* **cert_path**: Path to a certificate file in PEM or PKCS12 format, including the private key.
+* **client_send_cert_chain**: (Optional) If True, the credential will send the public certificate chain in the x5c header of each token request's JWT. This is required for Subject Name/Issuer (SNI) authentication. Defaults to `false`.
+
+Assertion Authentication
+------------------------
+ClientAssertionCredential allows authentication using a pre-obtained JWT (Json Web Token) assertion instead of a client secret or certificate. It is primarily used for Service Principal authentication, but can also work with other identities if a valid JWT is provided.
+
+Example:
+.. code:: yaml
+
+   platform:
+     - type: azure
+       azure:
+         credential:
+           type: assertion
+           tenant_id: <tenant id> # Required
```

Comment:
It still has the client_id, because the client_id is in base class.

Reply:
client id is not used in this authentication way